### ISSUE TITLE
[RobotServer/FakeMotorController] dummy option always found

### ISSUE THREAD

**Issue by jgvictores** [asrob-uc3m/yarp-devices#12]
[RobotServer/FakeMotorController] dummy option always found, current `develop` (ec5f5a4c20e81ee225945e719d52f7379e4a5642), replicable via:
```
yarpdev --device FakeMotorController --verbose
```
Gets:
```
[info] DeviceDriverImpl.cpp:17 open(): Subdevice FakeMotorController
[debug] DeviceDriverImpl.cpp:12 open(): Got options: (device FakeMotorController) (dummy 123) (single_threaded 1) (subdevice FakeMotorController) (verbose)
[success] DeviceDriverImpl.cpp:16 open(): Got dummy option.
[INFO]created device <FakeMotorController>. See C++ class asrob::FakeMotorController for documentation.
yarp: Port /RobotServer/rpc:s active at tcp://163.117.150.234:10002/
[INFO]created wrapper <RobotServer>. See C++ class asrob::RobotServer for documentation.
===============================================================
== Options checked by device:
== 
device=FakeMotorController
RobotServer.subdevice=FakeMotorController
FakeMotorController.device=FakeMotorController
FakeMotorController.dummy
    check dummy option
RobotServer.name
==
===============================================================
yarp: Port /FakeMotorController/quit active at tcp://163.117.150.234:10003/
[INFO]device active in background...
```
Maybe related to https://github.com/roboticslab-uc3m/yarp-devices/issues/175 ? (possibly not)

**Comment by PeterBowman** [comment]
Can't replicate this. Also:

> (dummy 123)

This bottle does not show up on my console output upon `yarpdev --device FakeMotorController --verbose`, I guess you passed on the `--dummy 123` option.

**Comment by jgvictores** [comment]
Mmm,  sorry for the bad copy+paste.

```
yarpdev --device FakeMotorController
[info] DeviceDriverImpl.cpp:17 open(): Subdevice FakeMotorController
[debug] DeviceDriverImpl.cpp:12 open(): Got options: (device FakeMotorController) (single_threaded 1) (subdevice FakeMotorController)
[success] DeviceDriverImpl.cpp:16 open(): Got dummy option.
[INFO]created device <FakeMotorController>. See C++ class asrob::FakeMotorController for documentation.
yarp: Port /RobotServer/rpc:s active at tcp://163.117.150.234:10002/
[INFO]created wrapper <RobotServer>. See C++ class asrob::RobotServer for documentation.
yarp: Port /FakeMotorController/quit active at tcp://163.117.150.234:10003/
[INFO]device active in background...
```

```
yarpdev --device FakeMotorController --verbose
||| configuring
[info] DeviceDriverImpl.cpp:17 open(): Subdevice FakeMotorController
[debug] DeviceDriverImpl.cpp:12 open(): Got options: (device FakeMotorController) (single_threaded 1) (subdevice FakeMotorController) (verbose)
[success] DeviceDriverImpl.cpp:16 open(): Got dummy option.
[INFO]created device <FakeMotorController>. See C++ class asrob::FakeMotorController for documentation.
yarp: Port /RobotServer/rpc:s active at tcp://163.117.150.234:10002/
[INFO]created wrapper <RobotServer>. See C++ class asrob::RobotServer for documentation.
===============================================================
== Options checked by device:
== 
device=FakeMotorController
RobotServer.subdevice=FakeMotorController
FakeMotorController.device=FakeMotorController
FakeMotorController.dummy
    check dummy option
RobotServer.name
==
===============================================================
yarp: Port /FakeMotorController/quit active at tcp://163.117.150.234:10003/
[INFO]device active in background...
```

**Comment by jgvictores** [comment]
No `[success] DeviceDriverImpl.cpp:16 open(): Got dummy option.` line for you? :-(

**Comment by jgvictores** [comment]
Mmmm, okay, something is really wrong... commenting out setMonitor it also happens.

**Comment by jgvictores** [comment]
Reinstalling YARP, will comment later on.

**Comment by PeterBowman** [comment]
Oops, sorry, I got fooled by the weird color output on Windows :(. Yes, I
see that, too.

**Comment by PeterBowman** [comment]
The following check evaluates to true no matter what option key is passed (`dummy`, `dummy2`, `hola`...):

https://github.com/asrob-uc3m/yarp-devices/blob/ec5f5a4c20e81ee225945e719d52f7379e4a5642/libraries/YarpPlugins/FakeMotorController/DeviceDriverImpl.cpp#L14-L17

However, the one-parameter overload works as expected: `if( config.check("whatever") ) {}`.

**Comment by PeterBowman** [comment]
@jgvictores haven't you noticed that stray semicolon? 😆

**Comment by jgvictores** [comment]
OMG

**Comment by jgvictores** [comment]
Hehehehehe thanks! Fixed at 36996bf46851dd3ba0d7dba90ad8737671732e5f
